File: src/model/device.js
var validate = require('../utils/validate');

class Device{

    constructor(deviceID, energyUsage){
        this.deviceID = deviceID;
        this.energyUsage = energyUsage

        this.validate();
    }

    toDTO(){
        return {
            DeviceId: this.deviceID,
            EnergyUsage: this.energyUsage
        };
    }

    validate(){
        validate.string(this.deviceID, "DeviceId");
        validate.number(this.energyUsage, "EnergyUsage");
    }
}

module.exports.default = Device;

File: src/model/device.py
from src.utils import validate

class Device:

    #Constructor
    def __init__(self, deviceID, energyUsage):
        self.deviceID = deviceID
        self.energyUsage = energyUsage

     #   self.validate()

    #Serialization
    def serialize(self):
        return {
            "DeviceId": self.deviceID,
            "EnergyUsage": self.energyUsage
        }

    #Validating
    def validate(self):
        validate.string(self.deviceID, "DeviceId")
        validate.number(self.energyUsage, "EnergyUsage")
